pyomo/core/tests/unit/test_block.py

```diff
@@ -776,7 +778,7 @@ def __init__(self, *args, **kwds):
 
         b.clear()
         b.transfer_attributes_from(c)
-        self.assertEqual(list(b.component_map()), ['y','x','z'])
+        self.assertEqual(list(b.component_map()), ['y','z','x'])
```

Comment:
Was this a bug before? Why did it change?

Reply:
We used to sort the incoming dict to ensure determinism.  Now that we only support Python 3.7+, we can remove the sort and preserve decl order.  So not a bug per se, but this is a "better" approach (the user can actually control the decl order if they know what they are doing).